cmd/consensus.cpp

```diff
@@ -284,56 +293,58 @@ Status run_block(const nlohmann::json& b, const ChainConfig& config, IntraBlockS
         return kFailed;
     }
 
-    processor.evm().state().write_to_db(block_number);
-
     if (invalid) {
         std::cout << "Invalid block executed successfully\n";
         std::cout << "Expected: " << b["expectException"] << "\n";
         return kFailed;
     }
 
-    state.db().insert_header(block.header);
+    state.insert_header(block.header);
 
     return kPassed;
 }
 
-bool post_check(const IntraBlockState& state, const nlohmann::json& expected) {
+bool post_check(const StateBuffer& state, const nlohmann::json& expected) {
     for (const auto& entry : expected.items()) {
         evmc::address address{to_address(*from_hex(entry.key()))};
-        const nlohmann::json& account{entry.value()};
+        const nlohmann::json& j{entry.value()};
+
+        std::optional<Account> account{state.read_account(address)};
+        if (!account) {
+            std::cout << "Missing account " << entry.key() << "\n";
+            return false;
+        }
 
-        Bytes balance_str{*from_hex(account["balance"].get<std::string>())};
+        Bytes balance_str{*from_hex(j["balance"].get<std::string>())};
         auto [expected_balance, err1]{rlp::read_uint256(balance_str, /*allow_leading_zeros=*/true)};
         check_rlp_err(err1);
-        intx::uint256 actual_balance{state.get_balance(address)};
-        if (actual_balance != expected_balance) {
+        if (account->balance != expected_balance) {
             std::cout << "Balance mismatch for " << entry.key() << ":\n";
-            std::cout << to_string(actual_balance, 16) << " ≠ " << account["balance"] << "\n";
+            std::cout << to_string(account->balance, 16) << " ≠ " << j["balance"] << "\n";
```

Comment:
Please use ASCII chars as we get different display output on windows.
I'd go for `!=` instead of `≠`

Reply:
Done.